src/EditorFeatures/Core/Implementation/AutomaticCompletion/BraceCompletionSessionProvider.BraceCompletionSession.cs

```diff
@@ -364,6 +352,61 @@ private void MoveCaretToClosingPoint()
                 }
             }
 
+            private void ApplyBraceCompletionResult(BraceCompletionResult result)
+            {
+                using var edit = SubjectBuffer.CreateEdit();
+                foreach (var change in result.TextChanges)
+                {
+                    edit.Replace(change.Span.ToSpan(), change.NewText);
+                }
+
+                edit.ApplyAndLogExceptions();
+
+                Debug.Assert(SubjectBuffer.CurrentSnapshot[OpeningPoint.GetPosition(SubjectBuffer.CurrentSnapshot)] == OpeningBrace,
+                "The opening point does not match the opening brace character");
+                Debug.Assert(SubjectBuffer.CurrentSnapshot[ClosingPoint.GetPosition(SubjectBuffer.CurrentSnapshot) - 1] == ClosingBrace,
+                "The closing point does not match the closing brace character");
+
+                var snapshotPoint = SubjectBuffer.CurrentSnapshot.GetPoint(result.CaretLocation);
+                var line = snapshotPoint.GetContainingLine();
+
+                if (line.GetText().IsNullOrWhiteSpace())
+                {
+                    // If the caret is on an empty line, put it at the correct indentation using virtual space.
+                    PutCaretOnLine(this, _smartIndentationService, line.LineNumber);
```

Comment:
nit: it seems unfortunate to me that this value is not part of the BraceCompletionResult .  i would think *any* host would want this info to set the caret properly, and it should just be part of hte 'data' we compute and return.

Reply:
The caret location is returned - I can add an additional property for the indentation if on a new line